src/Platform/OpenGL/Shaders/dof_coc.glsl

```diff
@@ -0,0 +1,25 @@
+#include "Library/shader_utils.glsl"
+in vec2 TexCoord;
+out vec4 OutColor;
+uniform sampler2D albedoTex;
+uniform sampler2D normalTex;
+uniform sampler2D materialTex;
+uniform sampler2D depthTex;
+struct Camera
+{
+    vec3 position;
+    mat4 invViewProjMatrix;
+    mat4 viewProjMatrix;
+};
+uniform sampler2D cameraOutput;
+uniform Camera camera;
+uniform float focusDistance;
+uniform float focusRange;
+void main()
+{
+    FragmentInfo fragment = getFragmentInfo(TexCoord, albedoTex, normalTex, materialTex, depthTex, camera.invViewProjMatrix);
+    float fragDistance = length(fragment.position - camera.position);
+    float circleOfConfusion = (fragDistance - focusDistance) / focusRange;
+    circleOfConfusion = clamp(circleOfConfusion, -1., 1.);
+    OutColor = vec4(abs(circleOfConfusion), 0.f.xxx);
```

Comment:
utilizing just one channel of hdr rgba texture is quite a lot of memory waste I guess